### PR TITLE
deps(sentry): upgrade Sentry to version 8

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - Sentry 8 is the latest version

This commit:
 - deletes some integration code since Sentry does it automatically
 - updates code to use the Sentry 8 API
 - updates a few proxyquire calls because they were cause module not
   found errors in CI

### CODE REVIEW

File: packages/fxa-auth-server/test/oauth/routes/token.js [hunk 1]
```diff
@@ -23,7 +24,8 @@ const CODE_WITHOUT_KEYS = 'f0f0f0';
 const mockDb = { touchSessionToken: sinon.stub() };
 const mockStatsD = { increment: sinon.stub() };
 const mockGlean = { oauth: { tokenCreated: sinon.stub() } };
-const tokenRoutes = proxyquire('../../../lib/routes/oauth/token', {
+const tokenRoutePath = path.join(__dirname, '../../../lib/routes/oauth/token');
```
Comment (dschom) [inline packages/fxa-auth-server/test/oauth/routes/token.js:27]: Is this causing the error?

Comment (chenba) [reply packages/fxa-auth-server/test/oauth/routes/token.js:27]: That's the fix.

**Comment by chenba** [comment]
I'm putting this up for an initial review.  But at the time of writing (mid-Sept), we've exceed the tracing quota on all projects(?) but the content server, making verification difficult.

File: packages/fxa-auth-server/lib/sentry.js [hunk 2]
```diff
@@ -120,6 +115,7 @@ async function configureSentry(server, config, processName = 'key_server') {
       method(request, h) {
         request.sentryScope = new Sentry.Scope();
 
+        /**
```
Comment (chenba) [inline packages/fxa-auth-server/lib/sentry.js:118]: I commented out the code block below and didn't notice any difference.  But it's entirely possible I was not looking in the right place.  Maybe @dschom can let me know what I should look for.  🙇

File: libs/shared/sentry/src/lib/browser.ts [hunk 3]
```diff
@@ -166,12 +170,7 @@ function configure(config: SentryConfigOpts, log?: Logger) {
   try {
     Sentry.init({
       ...opts,
-      integrations: [
-        Sentry.browserTracingIntegration({
-          enableInp: true,
-        }),
-      ],
```
Comment (chenba) [inline libs/shared/sentry/src/lib/browser.ts:173]: Most integrations are now automatic.

File: packages/fxa-auth-server/lib/sentry.js (hunk 2 above)
Comment (dschom) [reply packages/fxa-auth-server/lib/sentry.js:118]: Nice. That would indicate to me that the auto instrumentation is now capturing spans. I wonder if we even need this server extension at all anymore?

**Comment by chenba** [comment]
Setting the PR back to draft for now as we've noticed some performance issues with Sentry 8.x.

**Review by dschom — APPROVED** [review]

File: package.json
Comment (StaberindeZA) [inline package.json:1]: @chenba @dschom just wanted to mention the [resolutions of @sentry/types](https://github.com/mozilla/fxa/blob/74b337f50813426a8ad0ebb3a1be093d5dc98e99/package.json#L283). Seems like it's fine to remove that now.

**Comment by chenba** [comment]
This was broken into two PRs #17697 and #17698.  They've been merged so this can be closed without merging now.

**Pull request closed without merging**